packages/http/src/responses.ts

```diff
@@ -216,7 +214,7 @@ function getResponseDescription(
   // as the response description. This allows more freedom to change how
   // TypeSpec is expressed in semantically equivalent ways without causing
   // the output to change unnecessarily.
-  if (body === undefined || body.property) {
+  if (isResponseEnvelope(metadata)) {
```

Comment:
Can you look at how this impacts swagger in the azure-rest-api-specs repo?  This may require some spec updates or changes in higher-level libraries